### PR TITLE
Create a widget tracker service

### PR DESCRIPTION
**Pull request by blink1073**
cc @ellisonbg @jasongrout, based on conversations yesterday.

Adds a widget tracker service that will be upgraded once we have new APIs for widget semantic focus tracking.

### CODE REVIEW

File: src/widgettracker/index.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    return this._widgets = null;
+  }
+
+  /**
+   * The read-only list of tracked widgets.
+   *
+   * #### Notes
+   * This is a read-only property.
+   */
+  get widgets(): T[] {
```
Comment (ellisonbg) [inline src/widgettracker/index.ts]: Does the caller of this need to check to make sure none of the widgets have been disposed?

**Comment by ellisonbg** [comment]
Read through in a good amount of detail and things look great. Left just one question inline. Otherwise merge away!

File: src/widgettracker/index.ts (hunk 1 above)
Comment (sccolbert) [inline src/widgettracker/index.ts]: Disposed widgets are automatically removed from the tracker. If the caller holds onto the returned array (a shallow copy), then yes.

**Comment by ellisonbg** [comment]
Looks great merging now.

**Pull request merged**